rip_music_unix.cpp

```diff
@@ -153,11 +153,11 @@ static char * find_dev(const char * mnt_dir) {
 	char * dev = NULL;
 	FILE * file;
 
-	file = setmntent("/etc/mtab", "r");
+	file = setmntent("/proc/self/mounts", "r");
```

Comment:
http://man7.org/linux/man-pages/man3/setmntent.3.html
> These routines are used to access the filesystem description file _/etc/fstab_ and the mounted filesystem description file _/etc/mtab_.

There is no evidence about _/proc/self/mounts_ nor that **setmntent** can be used with different file.

Reply:
It's dropped from util-linux in 2.27 or so. You can read explanation [here](https://git.kernel.org/pub/scm/utils/util-linux/util-linux.git/commit/?id=e778642a9eb96975fcf3baa61dfa10add628af1a).
And in mainstream distributions `/etc/mtab` is a symlink to `/proc/mounts` or `/proc/self/mounts` anyway.

Reply:
But documentation for setmntent says something different... So who is right (and why)? :-)

Reply:
Distributions maintainers are right. It's related to namespaces and containers usage. You can just read `man mount` or anything  using libmount:
>The programs mount and umount traditionally maintained a list of currently mounted filesystems in the file /etc/mtab.  This real mtab file is still supported, but on current Linux systems  it is better to make it a symlink to /proc/mounts instead, because a regular mtab file maintained in userspace cannot reliably work with namespaces, containers and other advanced Linux features.

Reply:
Should not be then documentation for `setmntent` updated?

Reply:
I don't know, maybe it should, or maybe the actual source of data doesn't matter in scope of this function.

Anyway, the problem is that `/etc/mtab` file may not be present at all. For example, Flatpak's freedesktop.org runtime (used as root filesystem for an application running inside sandbox) doesn't provide `/etc/mtab` even as a compat symlink.